devito/jit_manager.py

```diff
@@ -45,7 +45,14 @@ def __init__(self, propagators, dtype=None, openmp=False):
             self._incompatible_flags = self._incompatible_flags + flags
 
         function_descriptors = [prop.get_fd() for prop in propagators]
+
         self.function_manager = FunctionManager(function_descriptors, self._mic_flag, self._openmp)
+
+        # add main method for auto tuning to produce executable file
+        if propagators[0].auto_tune and propagators[0].cache_blocking:
```

Comment:
This introduces an unnecessary dependency between the `JitManager` and the `Propagator`. This could be added as a flag to the `FunctionDescriptor` and this way the JitManager does not need to be aware of how the internals of the Propagator work, and neither the details of which functions need an associated `main` function.